.github/workflows/upload-pypi-source.yml

```diff
@@ -129,6 +129,10 @@ jobs:
             command: flake8 benchmarks build_scripts chia tests tools *.py
           - name: pylint
             command: pylint benchmarks build_scripts chia tests tools *.py
+          - name: generated protocol tests
```

Comment:
Thoughts about just making this a pre-commit step?

Reply:
not really. the main purpose of these tests are to make sure we don't accidentally change a network protocol message, which would make it incompatible with previous versions. I think it's good to have an extra manual step to ensure you really mean it in this case.

Reply:
I don't really understand what you want to tell me with this know :) I got the purpose of those tests, my question was actually about running this CI check you introduce here in `pre-commit` instead of `upload-pypi-source.yml`. Maybe i miss some difference here but in my mind running it in `pre-commit` has the same outcome on CI than what you do now + it already fails locally if you miss to rebuild the files (which should be done fast enough to not bother on `pre-commit` locally?). Or what is the purpose of this new step here?

Reply:
It got a new workflow, I suggested this instead.  Could go in pre-commit, but we all know I'm not a huge fan.  Not that I need to argue against it going there.

Reply:
> It got a new workflow, I suggested this instead.

Which new workflow? 

> Could go in pre-commit, but we all know I'm not a huge fan. Not that I need to argue against it going there.

Your not really a fan of pre-commit for _some_ reason yeah, but now this introduces the inconsistency to run our "CI check tools" in two different places while we get one more distinctions between local and CI linting which is not critical but also not heavy and could just run locally also.

Reply:
I'm fine with it going in pre-commit.  Had a longer keybase chat with Kevin as well.